requirements.txt

```diff
@@ -11,3 +11,4 @@ torchvision==0.3.0
 websocket_client>=0.56.0
 websockets>=7.0
 zstd>=1.4.0.0
+git+git://github.com/OpenMined/proto@master
```

Comment:
Sure, but there's not really a need to have this be a public package.  It's not a project that's useful in any open-source context and there's no problems with installing via git.  For me, I'd be in favor of doing things the way they are currently in this PR, it's also the approach we took in syft.js for the above reason.

Reply:
The other issue here @robert-wagner is that we will ideally be publishing updates to proto according to PySyft deployments.  Whenever a new PySyft is deployed, it should kick off a new proto deployment.  This is going to be done in the future.

However, adding this package to PyPi and NPM would only complicate the matter further.  It would also make it harder for syft.js and other libraries to stay up to date.  Can we avoid doing this for now until a better situation presents itself in the future?  @robert-wagner

Reply:
But what if i wanted an old version of proto? Dos this allow for that?

Reply:
All for it assuming we can reference old versions of proto. Working on automating as much of the deployment as possible so we should not worry about the pypi/npm problems going forward

Reply:
We can create tags in proto's git and reference specific tag in PySyft's requirements.txt. There can be extra step of publishing these tags to pypi/npm but it's not as important as figuring out the versioning process itself (i.e. which version to assign and how to sync pysyft and proto).
@robert-wagner what is PySyft publishing procedure currently?
Perhaps the simplest plan could be PySyft@dev using proto@master and then cut proto versions in sync with PySyft. Is it possible to have `proto@master` in requirements.txt of PySyft@dev, but freeze proto version in requirements.txt when publishing PySyft?
For example:
1. bumpversion on pysyft@dev (0.1.28a1 -> 0.1.29a1)
2. make pysyft branch `release-0.1.29a1`
3. [in release-0.1.29a1 branch] update requirements.txt: proto@master -> proto@0.1.29a1
4. tag pysyft@release-0.1.29a1 HEAD as pysyft@0.1.29a1
5. tag proto@master HEAD as proto@0.1.29a1
6. publish pysyft@0.1.29a1

Reply:
@iamtrask brings up a good point.  I think it would be wise to do some sort of versioning process.  @vvmnnnkv and I will determine some sort of solution and report back.

Reply:
We absolutely could do something like described. I will be working on automating a lot of the release processes this week. As of right now as long as you document the processes that you want, I can come back to it later and get it standardized (it should be pretty similar to much of the other work needed for syft-tf)

Reply:
@robert-wagner Please see suggested dev workflow regarding updates that involve proto:
https://github.com/OpenMined/PySyft/pull/2439/commits/7a19d7cd33ea647e9a518edbfe075077a070eeb6
That way PySyft@dev can continue using github dependency of proto@master (to avoid publishing proto every time somebody makes a change between pysyft releases). The only concern is who will do the last step (maybe some automation is possible?).

For PySyft release, I think it can be same as described above:
1. At the time of PySyft release, proto@master is tagged with PySyft version
2. PySyft's requirements.txt is updated with tagged proto version
3. PySyft is released/published

There may be a step to publish proto to pypi between 1 and 2. In this case requirements.txt is updated with reference to this version instead of reference to github tag.